Patch for kadmin kdb5_util default realm contributed by Matt Rogers.

--- a/src/kadmin/server/ipropd_svc.c	
+++ a/src/kadmin/server/ipropd_svc.c	
@@ -248,6 +248,7 @@ ipropx_resync(uint32_t vers, struct svc_req *rqstp)
     static kdb_fullresync_result_t ret;
     char *tmpf = 0;
     char *ubuf = 0;
+    char *realmbuf = 0;
     char clhost[MAXHOSTNAMELEN] = {0};
     int pret, fret;
     kadm5_server_handle_t handle = global_server_handle;
@@ -332,14 +333,24 @@ ipropx_resync(uint32_t vers, struct svc_req *rqstp)
 	goto out;
     }
 
+    if (handle->params.realm) {
+	if (asprintf(&realmbuf, "-r %s ", handle->params.realm) < 0) {
+	    krb5_klog_syslog(LOG_ERR,
+			     _("%s: unable to construct kdb5_util realm option; out of memory"),
+			     whoami);
+	    goto out;
+	}
+    }
+
     /*
      * note the -i; modified version of kdb5_util dump format
      * to include sno (serial number). This argument is now
      * versioned (-i0 for legacy dump format, -i1 for ipropx
      * version 1 format, etc)
      */
-    if (asprintf(&ubuf, "%s dump -i%d %s </dev/null 2>&1",
-		 KPROPD_DEFAULT_KDB5_UTIL, vers, tmpf) < 0) {
+    if (asprintf(&ubuf, "%s %sdump -i%d %s </dev/null 2>&1",
+		 KPROPD_DEFAULT_KDB5_UTIL, (realmbuf) ? realmbuf : "", vers,
+		 tmpf) < 0) {
 	krb5_klog_syslog(LOG_ERR,
 			 _("%s: cannot construct kdb5 util dump string too long; out of memory"),
 			 whoami);
--- a/src/slave/kpropd.c	
+++ a/src/slave/kpropd.c	
@@ -657,6 +657,12 @@ krb5_error_code do_iprop(kdb_log_context *log_ctx)
         }
     }
 
+    if (retval = krb5_set_default_realm(kpropd_context, def_realm)) {
+       com_err(progname, retval,
+               _("Unable to set default realm"));
+       exit(1);
+    }
+
     params.mask |= KADM5_CONFIG_REALM;
     params.realm = def_realm;
 
